client/lib/form-base/reducer.js

```diff
@@ -0,0 +1,45 @@
+import { SET_FORM_DATA_VALUE, SET_FORM_META_PROPERTY } from './actions';
+
+export const initialState = {
+	meta: {
+		error: false,
+		fieldsStatus: false,
+		isSaving: false,
+		pristine: true,
+	},
+	data: {
+	},
+};
+
+const reducers = {};
+
+reducers[ SET_FORM_DATA_VALUE ] = ( state, action ) => {
+	const data = Object.assign( {}, state.data, { [ action.key ]: action.value } );
+	const meta = Object.assign( {}, state.meta, { pristine: false } );
+	return {
+		meta: meta,
+		data: data,
+	};
+};
+
+reducers[ SET_FORM_META_PROPERTY ] = ( state, action ) => {
+	const data = Object.assign( {}, state.data );
+	const meta = Object.assign( {}, state.meta, { [ action.key ]: action.value } );
+	return {
+		meta: meta,
+		data: data,
+	};
+};
+
+const reducer = ( state, action ) => {
+	if ( 'undefined' === typeof state ) {
```

Comment:
You can declare this function like this instead, and remove the `undefined` check:
`const reducer = ( state = initialState, action ) => {`
See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters#Examples